src/org/rascalmpl/uri/URIResolverRegistry.java

```diff
@@ -1039,16 +1039,17 @@ private void mkParentDir(ISourceLocation uri) throws IOException {
 	}
 
 	public void watch(ISourceLocation loc, boolean recursive, final Consumer<ISourceLocationChanged> callback) throws IOException {
-		if (getOutputResolver(safeResolve(loc).getScheme()) == null) {
-			throw new UnsupportedSchemeException("Watching not supported on schemes that do not support writing: " + loc.getScheme());
-		}
-		watchers.watch(loc, recursive, callback);
+		watchers.watch(loc, recursive, this::anyResolverRegistered , callback);
+	}
+
+	private boolean anyResolverRegistered(ISourceLocation loc) {
```

Comment:
This name might be confusing, as this function only tests for availability of an input or output resolver (and not logical or classloader resolver)

Reply:
thanks, I've improved it.